internal/db/db.go

```diff
@@ -14,9 +14,27 @@ type DB interface {
 	// ListTools returns all tools. Returns nil if no tools were found, error will
 	// be non-nil if an error occurs.
 	ListTools() ([]Tool, error)
+	// StartAnalysis records a new analysis.
+	StartAnalysis(ghInstallationID, repositoryID int) (analysisID int, err error)
+	// FinishAnalysis marks a status as finished.
+	FinishAnalysis(analysisID int, status AnalysisStatus, analysis *Analysis) error
+	// ErrorAnalysis marks a status as an (internal) error.
+	//ErrorAnalysis(analysisID int) error
 }
 
+// AnalysisStatus represents a status in the builds table.
+type AnalysisStatus string
+
+const (
+	AnalysisStatusPending AnalysisStatus = "Pending" // AnalysisStatusPending is a pending/started analysis.
```

Comment:
Wanna find an existing issue or file a new one?

I'm finding these somewhat related ones (but not quite exactly it) so far:

- https://github.com/golang/lint/issues/275
- https://github.com/golang/lint/issues/208